assets/js/base/context/cart-checkout/checkout/index.js

```diff
@@ -41,14 +41,14 @@ export const CheckoutProvider = ( {
 			submitLabel={ submitLabel }
 		>
 			<BillingDataProvider>
-				<PaymentMethodDataProvider
-					activePaymentMethod={ initialActivePaymentMethod }
-				>
-					<ShippingDataProvider>
+				<ShippingDataProvider>
+					<PaymentMethodDataProvider
+						activePaymentMethod={ initialActivePaymentMethod }
+					>
 						{ children }
 						<CheckoutProcessor />
-					</ShippingDataProvider>
-				</PaymentMethodDataProvider>
+					</PaymentMethodDataProvider>
+				</ShippingDataProvider>
```

Comment:
Due to the PaymentMethodDataProvider now consuming shipping data context (so `setPaymentStatus.success()` also receives shipping data to update in the state), I had to change the nesting order of the providers.

Reply:
Does it make sense to nest in order of fields on checkout? Shipping>Billing>Payment? Does billing and shipping being nested potentially cause issues down the road if they depend on each other?

Reply:
Payment is fine as is. We may need to switch Billing and Shipping around but that can be determined in followup?  I think the only place where it's potentially problematic is for copying data between shipping and billing (in which case the current order should be sufficient right (because shipping needs to write to billing).